src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -85,133 +125,145 @@ public bool IsSupported(FunctionMetadata functionMetadata, string workerRuntime)
                 }
             }
         }
 
-        private LanguageWorkerState CreateWorkerState(string runtime)
+        public async Task InvokeAsync(ScriptInvocationContext invocationContext)
         {
-            var state = new LanguageWorkerState();
-            WorkerConfig config = _workerConfigs.Where(c => c.Language.Equals(runtime, StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
-            state.Channel = ChannelFactory(runtime, state.Functions, 0);
-            _workerStates[runtime] = state;
-            return state;
+            if (_workerState.ProcessRestartCountExceedException != null)
+            {
+                invocationContext.ResultSource.TrySetException(_workerState.ProcessRestartCountExceedException);
+                return;
+            }
+            try
+            {
+                await Utility.InvokeWithRetriesAsync(() =>
+                {
+                    IEnumerable<ILanguageWorkerChannel> workerChannels = GetInitializedWorkerChannels();
+                    var languageWorkerChannel = _functionDispatcherLoadBalancer.GetLanguageWorkerChannel(workerChannels, _maxProcessCount);
+                    BufferBlock<ScriptInvocationContext> bufferBlock = null;
+                    if (languageWorkerChannel.FunctionInputBuffers.TryGetValue(invocationContext.FunctionMetadata.FunctionId, out bufferBlock))
+                    {
+                        _logger.LogDebug($"Posting invocation id:{invocationContext.ExecutionContext.InvocationId}");
+                        languageWorkerChannel.FunctionInputBuffers[invocationContext.FunctionMetadata.FunctionId].Post(invocationContext);
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException($"Function:{invocationContext.FunctionMetadata.Name} not loaded by the language worker with workerId:{languageWorkerChannel.Id}");
+                    }
+                }, maxRetries: 2, retryInterval: TimeSpan.FromMilliseconds(100));
+            }
+            catch (Exception invokeEx)
+            {
+                invocationContext.ResultSource.TrySetException(invokeEx);
+            }
         }
 
-        public void Register(FunctionRegistrationContext context)
+        internal IEnumerable<ILanguageWorkerChannel> GetInitializedWorkerChannels()
         {
-            _workerStates[context.Metadata.Language].Functions.OnNext(context);
+            var webhostChannels = _languageWorkerChannelManager.GetChannels(_workerRuntime);
+            var workerChannels = webhostChannels == null ? _workerState.GetChannels() : webhostChannels.Union(_workerState.GetChannels());
+            return workerChannels.Where(ch => ch.State == LanguageWorkerChannelState.Initialized);
         }
 
-        public void WorkerError(WorkerErrorEvent workerError)
+        public async void WorkerError(WorkerErrorEvent workerError)
```

Comment:
Check if this is disposing before restarging worker channel

Reply:
done!